reference/normalize.go

```diff
@@ -1,9 +1,124 @@
 package reference
 
+import (
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/docker/distribution/digest"
+)
+
 var (
-	defaultTag = "latest"
+	legacyDefaultDomain = "index.docker.io"
+	defaultDomain       = "docker.io"
+	defaultRepoPrefix   = "library/"
+	defaultTag          = "latest"
```

Comment:
Any thoughts on exporting these? If we don't, downstream code will likely need to redefine them.